ios/Classes/AblyFlutterPlugin.m

```diff
@@ -39,6 +39,44 @@ -(nullable AblyFlutter *)ablyWithHandle:(NSNumber *)handle;
     [plugin registerWithCompletionHandler:result];
 };
 
+static FlutterHandler _createRestWithOptions = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
+    AblyFlutterMessage *const message = call.arguments;
+    LOG(@"message for handle %@", message.handle);
+    AblyFlutter *const ably = [plugin ablyWithHandle:message.handle];
+    // TODO if ably is nil here then an error response, perhaps? or allow Dart side to understand null response?
+    result([ably createRestWithOptions:message.message]);
+};
+
+static FlutterHandler _publishRestMessage = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
+    AblyFlutterMessage *const message = call.arguments;
+    LOG(@"message for handle %@", message.handle);
+    AblyFlutter *const ably = [plugin ablyWithHandle:message.handle];
+    AblyFlutterMessage *const messageData = message.message;
+    NSMutableDictionary<NSString *, NSObject *>* _dataMap = messageData.message;
+    NSString *const channelName = (NSString*)[_dataMap objectForKey:@"channel"];
+    NSString *eventName = (NSString*)[_dataMap objectForKey:@"name"];
+    NSObject *eventData = (NSString*)[_dataMap objectForKey:@"message"];
+    ARTRest *const client = [ably getRest:messageData.handle];
+    ARTRestChannel *const channel = [client.channels get:channelName];
+
+    //Handling nil cases, to be generalized based on other encounters
+    NSString *nilStr = nil;
+    if([eventName isKindOfClass:[NSNull class]]){ eventName = nilStr; }
+    if([eventData isKindOfClass:[NSNull class]]){ eventData = nilStr; }
+    [channel publish:eventName data:eventData callback:^(ARTErrorInfo *const error){
+        if(error){
+            result([
+                    FlutterError
+                    errorWithCode:[NSString stringWithFormat: @"%ld", (long)error.code]
+                    message:[NSString stringWithFormat:@"Unable to publish message to Ably server; err = %@", [error message]]
```

Comment:
@mattheworiordan @QuintinWillison @paddybyers 

Should we have AblyException representing ErrorInfo instead of encapsulating it?

So, `AblyException` will follow `ErrorInfo` spec but will implement `Exception`!

Reply:
If more context is what's being sought then, from an Objective-C perspective there's the opportunity to create error instances like this using a macro which then allows us to present deeper native context (e.g. `__FILE__` and `__LINE__`).

However, I'm not sure that's necessary. And, in fact, in this scenario we need to be just aiming to make the wrapper as thin as possible. Therefore I think @tiholic is proposing the right approach, being representation over composition. Might it be worth giving that a go first and seeing how that looks?

Reply:
> If more context is what's being sought then

Nope, I am stating that wrapping errors with errors, when the underlying error is likely to have a similar message, will make messages redundant in a lot of cases.  How do we approach this problem more generally with wrappers?

> Therefore I think @tiholic is proposing the right approach, being representation over composition. Might it be worth giving that a go first and seeing how that looks?

Not entirely clear what you're proposing, so happy to see.

Reply:
> Should we have AblyException representing ErrorInfo instead of encapsulating it?

I don't like that personally. I prefer that `ErrorInfo` is a PODO. If the language then requires us to wrap it in order to throw it, then we can do that, and the intent is clear to everyone.

Reply:
I want to address this sooner than later but we also have a backlog of PRs to get merged, this one being at the head, so for the moment I have created issue #10 in order to ensure that this doesn't get overlooked once we merge this PR.